### ISSUE TITLE
1M microphysics never publishes r_eff_liq, so echam.yaml runs on a constant 11 um droplet radius

### ISSUE THREAD

**Issue by duncanwp** [climate-analytics-lab/jax-gcm#717]
Split out of #670, which is reduced to dead-code cleanup — see [that comment](https://github.com/climate-analytics-lab/jax-gcm/issues/670#issuecomment-5383540879) for why the land-fraction half of it should be deleted rather than fixed.

## What is missing

`Echam1MMicrophysics` never writes `clouds.r_eff_liq`. `cloud_data.py` zero-initialises it and only `lohmann_2m/scheme.py` ever populates it, so on the 1M path `rrtmgp.py` falls back to `effective_radius_liquid(cdnc_factor, land_fraction)` — which is `(14*(1-lf) + 8*lf) * cdnc_factor**(-1/3)` with `land_fraction` defaulting to 0.5 and never passed.

The result on `physics=echam`: **cloud droplet effective radius is 11 um everywhere**, modulated only by the column-scalar `cdnc_factor`. It is constant in the vertical and independent of liquid water content, so cloud shortwave optical depth is exactly proportional to LWP with no droplet-size feedback.

## Scope — low priority, and why

Every production configuration runs 2M and is unaffected:

| config | resolved `cloud_scheme` |
|---|---|
| `echam-jam` | 2m |
| `echam-jam-aci` | 2m |
| `echam-rrtmgp-2m` | 2m |
| `echam` | **1m** |

So this reaches only the plain `echam.yaml` preset. It is a real fidelity gap there, not a correctness bug elsewhere.

## Where the hooks already are

The 2M path already implements the reference law — `lohmann_2m/assembly.py:243-264`, ECHAM's `ncd_activ /= 0` branch:

```python
breadth = breadth_factor(cdnc)                       # Peng & Lohmann (2003) eq. 6
liq_radius_base = (3.0 / (4.0 * pi * c.rhow)) * cloud_liquid_in_cloud * air_density / cdnc
liq_eff_radius = 1.0e6 * breadth * liq_radius_base ** (1.0 / 3.0)
```

**The fix is to call the same code, not to write a new formula.** Factor those lines into a shared helper (natural home `clouds/cloud_utils.py`, next to `breadth_factor`) and have both schemes use it.

The 1M term has everything it needs already:

- **CDNC**: `echam_1m.py` builds `cdnc_m3 = base_cdnc * cdnc_factor` from `diagnostics["aerosol"]` and already publishes it as `clouds.droplet_number` — the same array can feed the radius.
- **In-cloud LWC**: `micro_state.qc_in_cloud`. Note there is a real choice here between the input-state value (what ECHAM's radiation sees, `xlm1`) and the post-microphysics `qc_end` (what the 2M path uses) — decide it explicitly.
- **Publish mechanism**: add `r_eff_liq=<(nlev,ncols)>` to the existing `clouds.copy(...)` block, mirroring `lohmann_2m/scheme.py:1174-1180`. `CloudData.r_eff_liq` already exists, so no struct change.

Two things that are load-bearing if this is implemented:

- **cloud-free levels must stay exactly 0.0** — `rrtmgp.py` selects with `jnp.where(r_eff_liq_um > 0.0, ...)`, so the `jnp.where(has_liquid, ..., 0.0)` mask in the 2M code is not cosmetic;
- **copy the 2M double-`where` gradient guard verbatim** (`assembly.py:247-262`) — the cube root's derivative is infinite at 0 and the resulting NaN only appears from step 2 of a rollout, once radiation consumes the carry.

## Blast radius if implemented

Measured on 80,000 real training columns, ECHAM law at N = 100 cm-3 with RRTMGP's LUT clip [2.5, 21.5] um: median per-column cloud optical depth ratio **2.39x** (p5 0.69, p95 4.40), mean cloud albedo 0.434 -> 0.542, 22% of cloudy layers pinned at the 2.5 um floor. So this is a climate-changing change to `physics=echam` and needs its own validation, not a drive-by.

## Reference

ECHAM6 `mo_cloud_optics.f90`: `re = max(relmin, min(relmax, zfact*zkap*(zlwc/zcdnc)**(1/3)))` with `zfact = 1e6*(3e-9/(4*pi*rhoh2o))**(1/3)` = 62.035 um, `zlwc` in-cloud g/m3, `zcdnc` in cm-3. Martin et al. (1994) for the breadth parameter; Peng & Lohmann (2003) eq. 6 for its CDNC-dependent form.

**Comment by duncanwp** [comment]
Fixed by PR #730 (commit 3914226): the 1M scheme publishes r_eff_liq via the ECHAM preffl formulation, verified in that PR's validation. The PR said 'Refs' rather than 'Closes', so the issue never auto-closed — closing now. The residual mixed-phase r_eff_ice mechanism was split to #728 and stays open.
